### PR TITLE
Add use of Composer scripts

### PR DESCRIPTION
**Pull request by RobLoach**
What are your thoughts on updating the *Testing* section of the readme to document testing via `composer test`?

### CODE REVIEW

**Comment by reinink** [comment]
I for one think using `composer test` is a great idea. As seen in #21, simply running `phpunit` isn't necessarily always going to work. What if they don't have `vendor/bin` in their PATH? Or are we expecting everyone to have PHPUnit installed globally? And is that even a good recommendation?

Using [Composer commands](https://getcomposer.org/doc/articles/scripts.md#writing-custom-commands) solves all this. And if there are different testing frameworks being used, great, it will just work. This also takes care of the path resolution, removing the need to include things in your PATH to make the command work.

I'm pretty sure this is what @Seldaek was trying to solve with commands, so why not take advantage of this?

**Comment by reinink** [comment]
One more note on this pull request. You don't actually have to include the full path (`"test": "./vendor/bin/phpunit"`), since Composer automatically adds the bin, as per [the docs](https://getcomposer.org/doc/articles/scripts.md#writing-custom-commands):

> Note: Composer's bin-dir is pushed on top of the PATH so that binaries of dependencies are easily accessible as CLI commands when writing scripts.

**Comment by philsturgeon** [comment]
Right, for testing at work we use `rake test && dredd` which would be a pain for somebody else to know. You just want to run a single command and have it test.

The only other options are phing-style build managers and alias bash scripts which wont work on Windows, so Composer scripts are the obvious choice. :)

File: composer.json [hunk 1]
```diff
@@ -31,6 +31,9 @@
             "League\\Skeleton\\Test\\": "tests"
         }
     },
+    "scripts": {
+        "test": "./vendor/bin/phpunit"
```
Comment (Seldaek) [inline composer.json]: This should be only `"phpunit"`

Comment (philsturgeon) [inline composer.json]: Cheers! @reinink said the same.

**Comment by RobLoach** [comment]
Should we update it to use `phpunit --coverage-text --coverage-clover=coverage.clover`? Is that a setting we could maintain in `phpunit.xml.dist`?

**Comment by potherca** [comment]
You could use the `<logging>` section in the XML file. 
If I understand you correctly, this is what you want:

    <logging>
        <log type="coverage-text" target="php://stdout"/>
        <log type="coverage-clover" target="coverage.clover" />
    </logging>

I've had a long weekend, so I'm slightly error-prone today :smile:

**Comment by philsturgeon** [comment]
I thought about it long and hard and decided I was right. :p 

But seriously, this is a good change and I wish more people did it.

**Comment by RobLoach** [comment]
@philsturgeon Should we update `README.md` accordingly as well?

**Comment by philsturgeon** [comment]
Yessir please PR

**Comment by potherca** [comment]
@philsturgeon I used to be against this but all of the logic and sanity dictates this is a smart idea.
:+1: for merging this. I'll probably be advocating using `composer test` over other harder to remember and often changing commands across projects :smile_cat:

**Pull request merged**